{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "586ff918",
   "metadata": {},
   "source": [
    "# CMAP analysis\n",
    "\n",
    "In this notebook we analyze available gene expressio data from the CMAP data set of U2OS cells in various overexpression conditions including those covered in our structural regulators. The idea is to better understand the clustering of the gene expression profiles and the corresponding co-clustering with the currently inferred pyhsical space estimate. Additionally, we aim to preprocess the data such that it can be used later to infer the regulatory network of U2OS cells which will be the basis for the modelling of the regulatory space."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d11c1735",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "## 0. Environmental setup\n",
    "\n",
    "First, we setup the environment by loading required packages."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6bebf216",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from venn import venn\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from sklearn.cluster import AgglomerativeClustering, KMeans\n",
    "from cmapPy.pandasGEXpress.parse import parse\n",
    "\n",
    "import sys\n",
    "from yellowbrick.cluster import KElbowVisualizer\n",
    "\n",
    "sys.path.append(\"../../../..\")\n",
    "from src.utils.notebooks.ppi.embedding import plot_tsne_embs\n",
    "from src.utils.notebooks.images.embedding import *\n",
    "from src.utils.torch.network import *\n",
    "from src.utils.notebooks.ppi.embedding import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "185c711a",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "## 1. Read in data\n",
    "\n",
    "Next, we read the CMAP gene signature data and associated metadata."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6e9e7c1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-9-5a87311669de>:1: DtypeWarning: Columns (0,3,4,28,29) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  sig_info = pd.read_csv(\"../../../../data/resources/gex/cmap/siginfo_beta.txt\", sep=\"\\t\")\n"
     ]
    }
   ],
   "source": [
    "sig_info = pd.read_csv(\"../../../../data/resources/gex/cmap/siginfo_beta.txt\", sep=\"\\t\")\n",
    "gene_info = pd.read_csv(\n",
    "    \"../../../../data/resources/gex/cmap/geneinfo_beta.txt\", sep=\"\\t\"\n",
    ")\n",
    "cell_info = pd.read_csv(\n",
    "    \"../../../../data/resources/gex/cmap/cellinfo_beta.txt\", sep=\"\\t\"\n",
    ")\n",
    "l5_oe_gex_data = parse(\n",
    "    \"../../../../data/resources/gex/cmap/level5_beta_trt_oe_n34171x12328.gctx\"\n",
    ").data_df.transpose()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5173e3f5",
   "metadata": {},
   "source": [
    "We are particularly interested in the CMAP gene signatures corresponding to the overexpression of any of the genes contained in the inferred gene-gene interactome. Thus we also load the corresponding graph."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "663e7e5c",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "## 2. Data preprocessing\n",
    "\n",
    "To preprocess the data, we briefly filter out any signatures of the CMAP data set that are not related to any overexpression experiments related to our cell type of interest: U2OS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9ca31fd9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bead_batch</th>\n",
       "      <th>nearest_dose</th>\n",
       "      <th>pert_dose</th>\n",
       "      <th>pert_dose_unit</th>\n",
       "      <th>pert_idose</th>\n",
       "      <th>pert_itime</th>\n",
       "      <th>pert_time</th>\n",
       "      <th>pert_time_unit</th>\n",
       "      <th>cell_mfc_name</th>\n",
       "      <th>pert_mfc_id</th>\n",
       "      <th>...</th>\n",
       "      <th>cell_iname</th>\n",
       "      <th>det_wells</th>\n",
       "      <th>det_plates</th>\n",
       "      <th>distil_ids</th>\n",
       "      <th>build_name</th>\n",
       "      <th>project_code</th>\n",
       "      <th>cmap_name</th>\n",
       "      <th>is_exemplar_sig</th>\n",
       "      <th>is_ncs_sig</th>\n",
       "      <th>is_null_sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TA.OE005_U2OS_72H:BRDN0000464889:1</th>\n",
       "      <td>b15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>uL</td>\n",
       "      <td>1 uL</td>\n",
       "      <td>72 h</td>\n",
       "      <td>72.0</td>\n",
       "      <td>h</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>BRDN0000464889</td>\n",
       "      <td>...</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>O02</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15|TA.OE005_U2OS_72H_X2....</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15:O02|TA.OE005_U2OS_72H...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TA</td>\n",
       "      <td>SMAD3</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TA.OE005_U2OS_72H:BRDN0000407676:1</th>\n",
       "      <td>b15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>uL</td>\n",
       "      <td>1 uL</td>\n",
       "      <td>72 h</td>\n",
       "      <td>72.0</td>\n",
       "      <td>h</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>BRDN0000407676</td>\n",
       "      <td>...</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>I01</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15:I01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TA</td>\n",
       "      <td>WNT5A</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TA.OE005_U2OS_72H:BRDN0000464938:1</th>\n",
       "      <td>b15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>uL</td>\n",
       "      <td>1 uL</td>\n",
       "      <td>72 h</td>\n",
       "      <td>72.0</td>\n",
       "      <td>h</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>BRDN0000464938</td>\n",
       "      <td>...</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>J01</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15:J01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TA</td>\n",
       "      <td>HSP90B1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TA.OE005_U2OS_72H:BRDN0000459386:1</th>\n",
       "      <td>b15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>uL</td>\n",
       "      <td>1 uL</td>\n",
       "      <td>72 h</td>\n",
       "      <td>72.0</td>\n",
       "      <td>h</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>BRDN0000459386</td>\n",
       "      <td>...</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>G04</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15|TA.OE005_U2OS_72H_X2....</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15:G04|TA.OE005_U2OS_72H...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TA</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TA.OE005_U2OS_72H:BRDN0000399961:1</th>\n",
       "      <td>b15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>uL</td>\n",
       "      <td>1 uL</td>\n",
       "      <td>72 h</td>\n",
       "      <td>72.0</td>\n",
       "      <td>h</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>BRDN0000399961</td>\n",
       "      <td>...</td>\n",
       "      <td>U2OS</td>\n",
       "      <td>O13</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15|TA.OE005_U2OS_72H_X2....</td>\n",
       "      <td>TA.OE005_U2OS_72H_X1_B15:O13|TA.OE005_U2OS_72H...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TA</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   bead_batch  nearest_dose  pert_dose  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1        b15           NaN        1.0   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1        b15           NaN        1.0   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1        b15           NaN        1.0   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1        b15           NaN        1.0   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1        b15           NaN        1.0   \n",
       "\n",
       "                                   pert_dose_unit pert_idose pert_itime  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1             uL       1 uL       72 h   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1             uL       1 uL       72 h   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1             uL       1 uL       72 h   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1             uL       1 uL       72 h   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1             uL       1 uL       72 h   \n",
       "\n",
       "                                    pert_time pert_time_unit cell_mfc_name  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1       72.0              h          U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1       72.0              h          U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1       72.0              h          U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1       72.0              h          U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1       72.0              h          U2OS   \n",
       "\n",
       "                                       pert_mfc_id  ...  cell_iname  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1  BRDN0000464889  ...        U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1  BRDN0000407676  ...        U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1  BRDN0000464938  ...        U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1  BRDN0000459386  ...        U2OS   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1  BRDN0000399961  ...        U2OS   \n",
       "\n",
       "                                    det_wells  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1        O02   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1        I01   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1        J01   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1        G04   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1        O13   \n",
       "\n",
       "                                                                           det_plates  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1  TA.OE005_U2OS_72H_X1_B15|TA.OE005_U2OS_72H_X2....   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1                           TA.OE005_U2OS_72H_X1_B15   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1                           TA.OE005_U2OS_72H_X1_B15   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1  TA.OE005_U2OS_72H_X1_B15|TA.OE005_U2OS_72H_X2....   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1  TA.OE005_U2OS_72H_X1_B15|TA.OE005_U2OS_72H_X2....   \n",
       "\n",
       "                                                                           distil_ids  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1  TA.OE005_U2OS_72H_X1_B15:O02|TA.OE005_U2OS_72H...   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1                       TA.OE005_U2OS_72H_X1_B15:I01   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1                       TA.OE005_U2OS_72H_X1_B15:J01   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1  TA.OE005_U2OS_72H_X1_B15:G04|TA.OE005_U2OS_72H...   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1  TA.OE005_U2OS_72H_X1_B15:O13|TA.OE005_U2OS_72H...   \n",
       "\n",
       "                                    build_name project_code  cmap_name  \\\n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1         NaN           TA      SMAD3   \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1         NaN           TA      WNT5A   \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1         NaN           TA    HSP90B1   \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1         NaN           TA     PIK3CA   \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1         NaN           TA       TP53   \n",
       "\n",
       "                                    is_exemplar_sig  is_ncs_sig  is_null_sig  \n",
       "TA.OE005_U2OS_72H:BRDN0000464889:1                1         1.0          0.0  \n",
       "TA.OE005_U2OS_72H:BRDN0000407676:1                0         0.0          0.0  \n",
       "TA.OE005_U2OS_72H:BRDN0000464938:1                0         0.0          0.0  \n",
       "TA.OE005_U2OS_72H:BRDN0000459386:1                1         1.0          0.0  \n",
       "TA.OE005_U2OS_72H:BRDN0000399961:1                0         0.0          0.0  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_info = sig_info.loc[\n",
    "    (sig_info[\"project_code\"] == \"TA\") & (sig_info[\"cell_iname\"] == \"U2OS\")\n",
    "]\n",
    "sig_info = sig_info.loc[sig_info[\"pert_type\"].isin([\"trt_oe\"])]\n",
    "# sig_info = sig_info.loc[sig_info[\"cmap_name\"].isin(list(spearman_nodes))]\n",
    "sig_info.index = np.array(sig_info.sig_id)\n",
    "sig_info.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a506ede7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "324"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sig_info.cmap_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93667c37",
   "metadata": {},
   "source": [
    "We find 324 gene signatuures associated to the overexpression of 188 overexpressed genes."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62826d60",
   "metadata": {},
   "source": [
    "We will construct averaged gene sigantures for each overexpression conditions described above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f44a121e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-12-90bb33642f38>:5: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  l5_data = l5_oe_gex_data.loc[set(sig_info.sig_id)].copy()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>rid</th>\n",
       "      <th>NAT2</th>\n",
       "      <th>ADA</th>\n",
       "      <th>CDH2</th>\n",
       "      <th>AKT3</th>\n",
       "      <th>MED6</th>\n",
       "      <th>NAALAD2</th>\n",
       "      <th>NAALADL1</th>\n",
       "      <th>ACOT8</th>\n",
       "      <th>ABI1</th>\n",
       "      <th>GNPDA1</th>\n",
       "      <th>...</th>\n",
       "      <th>REC8</th>\n",
       "      <th>HNRNPDL</th>\n",
       "      <th>DMTF1</th>\n",
       "      <th>PPP4R1</th>\n",
       "      <th>CDH1</th>\n",
       "      <th>SLC12A6</th>\n",
       "      <th>PTBP3</th>\n",
       "      <th>KCNE2</th>\n",
       "      <th>DGCR2</th>\n",
       "      <th>SCO2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>labels</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACVR1B</th>\n",
       "      <td>0.382983</td>\n",
       "      <td>0.273100</td>\n",
       "      <td>0.556467</td>\n",
       "      <td>0.014800</td>\n",
       "      <td>0.108117</td>\n",
       "      <td>0.368000</td>\n",
       "      <td>0.534183</td>\n",
       "      <td>0.390683</td>\n",
       "      <td>-0.214333</td>\n",
       "      <td>0.201783</td>\n",
       "      <td>...</td>\n",
       "      <td>0.541300</td>\n",
       "      <td>-0.179367</td>\n",
       "      <td>-0.463917</td>\n",
       "      <td>0.169100</td>\n",
       "      <td>-1.299350</td>\n",
       "      <td>-0.120550</td>\n",
       "      <td>-0.093567</td>\n",
       "      <td>0.095350</td>\n",
       "      <td>0.599783</td>\n",
       "      <td>0.324317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADAM17</th>\n",
       "      <td>-0.321200</td>\n",
       "      <td>-0.866750</td>\n",
       "      <td>-0.070550</td>\n",
       "      <td>0.100850</td>\n",
       "      <td>-0.393200</td>\n",
       "      <td>1.122850</td>\n",
       "      <td>0.036250</td>\n",
       "      <td>0.468750</td>\n",
       "      <td>-0.743700</td>\n",
       "      <td>0.034900</td>\n",
       "      <td>...</td>\n",
       "      <td>0.345050</td>\n",
       "      <td>-0.369750</td>\n",
       "      <td>-0.322050</td>\n",
       "      <td>-0.210700</td>\n",
       "      <td>1.434200</td>\n",
       "      <td>0.058050</td>\n",
       "      <td>-0.212450</td>\n",
       "      <td>0.599150</td>\n",
       "      <td>-0.007250</td>\n",
       "      <td>1.124700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT1</th>\n",
       "      <td>-0.516100</td>\n",
       "      <td>-0.810000</td>\n",
       "      <td>0.099400</td>\n",
       "      <td>-0.500933</td>\n",
       "      <td>-0.105000</td>\n",
       "      <td>-0.560267</td>\n",
       "      <td>0.071867</td>\n",
       "      <td>0.357167</td>\n",
       "      <td>-0.135917</td>\n",
       "      <td>-0.618183</td>\n",
       "      <td>...</td>\n",
       "      <td>0.356483</td>\n",
       "      <td>0.234067</td>\n",
       "      <td>0.362783</td>\n",
       "      <td>0.904867</td>\n",
       "      <td>-0.072367</td>\n",
       "      <td>-0.444233</td>\n",
       "      <td>0.559650</td>\n",
       "      <td>0.015817</td>\n",
       "      <td>-0.102150</td>\n",
       "      <td>0.212633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT1S1</th>\n",
       "      <td>0.161675</td>\n",
       "      <td>-0.486725</td>\n",
       "      <td>0.648050</td>\n",
       "      <td>0.880625</td>\n",
       "      <td>-0.174125</td>\n",
       "      <td>1.154400</td>\n",
       "      <td>0.192925</td>\n",
       "      <td>0.297125</td>\n",
       "      <td>-0.207900</td>\n",
       "      <td>0.863800</td>\n",
       "      <td>...</td>\n",
       "      <td>0.080525</td>\n",
       "      <td>0.684025</td>\n",
       "      <td>-0.573075</td>\n",
       "      <td>0.495225</td>\n",
       "      <td>-1.047250</td>\n",
       "      <td>0.352625</td>\n",
       "      <td>0.582375</td>\n",
       "      <td>-0.536525</td>\n",
       "      <td>-0.496625</td>\n",
       "      <td>0.970000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKT2</th>\n",
       "      <td>-1.160950</td>\n",
       "      <td>0.428450</td>\n",
       "      <td>-0.063650</td>\n",
       "      <td>-1.065950</td>\n",
       "      <td>-0.131850</td>\n",
       "      <td>-0.108450</td>\n",
       "      <td>0.117800</td>\n",
       "      <td>-0.130300</td>\n",
       "      <td>-0.584200</td>\n",
       "      <td>0.209750</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.220450</td>\n",
       "      <td>1.569800</td>\n",
       "      <td>-1.108100</td>\n",
       "      <td>0.556550</td>\n",
       "      <td>0.411900</td>\n",
       "      <td>-0.621400</td>\n",
       "      <td>0.083950</td>\n",
       "      <td>-0.689700</td>\n",
       "      <td>-0.603000</td>\n",
       "      <td>1.102050</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 12328 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "rid         NAT2       ADA      CDH2      AKT3      MED6   NAALAD2  NAALADL1  \\\n",
       "labels                                                                         \n",
       "ACVR1B  0.382983  0.273100  0.556467  0.014800  0.108117  0.368000  0.534183   \n",
       "ADAM17 -0.321200 -0.866750 -0.070550  0.100850 -0.393200  1.122850  0.036250   \n",
       "AKT1   -0.516100 -0.810000  0.099400 -0.500933 -0.105000 -0.560267  0.071867   \n",
       "AKT1S1  0.161675 -0.486725  0.648050  0.880625 -0.174125  1.154400  0.192925   \n",
       "AKT2   -1.160950  0.428450 -0.063650 -1.065950 -0.131850 -0.108450  0.117800   \n",
       "\n",
       "rid        ACOT8      ABI1    GNPDA1  ...      REC8   HNRNPDL     DMTF1  \\\n",
       "labels                                ...                                 \n",
       "ACVR1B  0.390683 -0.214333  0.201783  ...  0.541300 -0.179367 -0.463917   \n",
       "ADAM17  0.468750 -0.743700  0.034900  ...  0.345050 -0.369750 -0.322050   \n",
       "AKT1    0.357167 -0.135917 -0.618183  ...  0.356483  0.234067  0.362783   \n",
       "AKT1S1  0.297125 -0.207900  0.863800  ...  0.080525  0.684025 -0.573075   \n",
       "AKT2   -0.130300 -0.584200  0.209750  ... -0.220450  1.569800 -1.108100   \n",
       "\n",
       "rid       PPP4R1      CDH1   SLC12A6     PTBP3     KCNE2     DGCR2      SCO2  \n",
       "labels                                                                        \n",
       "ACVR1B  0.169100 -1.299350 -0.120550 -0.093567  0.095350  0.599783  0.324317  \n",
       "ADAM17 -0.210700  1.434200  0.058050 -0.212450  0.599150 -0.007250  1.124700  \n",
       "AKT1    0.904867 -0.072367 -0.444233  0.559650  0.015817 -0.102150  0.212633  \n",
       "AKT1S1  0.495225 -1.047250  0.352625  0.582375 -0.536525 -0.496625  0.970000  \n",
       "AKT2    0.556550  0.411900 -0.621400  0.083950 -0.689700 -0.603000  1.102050  \n",
       "\n",
       "[5 rows x 12328 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_id_symbol_dict = dict(\n",
    "    zip(gene_info.loc[:, \"gene_id\"].astype(str), gene_info.loc[:, \"gene_symbol\"])\n",
    ")\n",
    "l5_oe_gex_data = l5_oe_gex_data.rename(gene_id_symbol_dict, axis=1)\n",
    "l5_data = l5_oe_gex_data.loc[set(sig_info.sig_id)].copy()\n",
    "l5_data[\"labels\"] = np.array(sig_info.loc[l5_data.index, \"cmap_name\"])\n",
    "mean_l5_data = l5_data.copy()\n",
    "mean_l5_data[\"labels\"] = np.array(sig_info.loc[mean_l5_data.index, \"cmap_name\"])\n",
    "mean_l5_data = mean_l5_data.groupby(\"labels\").mean()\n",
    "mean_l5_data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90150031",
   "metadata": {},
   "source": [
    "The gene signatures describe the differential gene expression profiles in the different overexpression settings. However, for only 978 genes the differential expression was computed based on measured quantities while the differential expression of the other genes contained in the signatures were inferred based on those 978 genes. Thus, we focus on the 978 landmark genes and subset the signatures accordingly to avoid any bias due to the inference of the expression of the other genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "878f6bfc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-13-63259ec9c24d>:2: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.\n",
      "  mean_l5_data = mean_l5_data.loc[:, l1000_genes]\n"
     ]
    }
   ],
   "source": [
    "l1000_genes = set(gene_info.loc[gene_info.feature_space == \"landmark\", \"gene_symbol\"])\n",
    "mean_l5_data = mean_l5_data.loc[:, l1000_genes]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29f126e1",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "## 4. Data export\n",
    "\n",
    "Finally, we export the CMap gene signatures for the selected nodes as well as the identified clustering structure."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c8e45d0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "mean_l5_data.to_csv(\"../../../../data/experiments/rohban/gex/cmap/mean_l5_signatures_tmp.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "206.167px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
